source/elements/oneMKL/source/domains/spblas/data_types/set_matrix_properties.rst

```diff
@@ -0,0 +1,107 @@
+.. SPDX-FileCopyrightText: 2024 Intel Corporation
+..
+.. SPDX-License-Identifier: CC-BY-4.0
+
+.. _onemkl_sparse_set_matrix_properties:
+
+set_matrix_properties
+=====================
+
+Matrix properties
+-----------------
+
+.. container:: section
+
+   .. code:: cpp
+
+      namespace oneapi::mkl::sparse {
+
+          enum matrix_property : std::int32_t {
+             symmetric = 1 << 0,
+             sorted    = 1 << 1,
+          };
+
+      }
+
+   Matrix properties are optional and "strong" guarantees. Unlike
+   :ref:`onemkl_sparse_matrix_view`, the user must ensure that the handle's data
+   holds all the given properties. A property can be set to optimize some
+   operations.
+
+   .. list-table::
+      :header-rows: 1
+      :widths: 20 80
+
+      * - Value
+        - Description
+      * - ``symmetric``
+        - Guarantees that the user-provided matrix data are symmetric, meaning
+          the matrix is square, the user data contain both lower and upper
+          triangular regions, and that its transpose is equal to itself.
+      * - ``sorted``
+        - | Guarantees that the user-provided matrix data has some sorting
+            property.
+          | For CSR this guarantees that the column indices are sorted in
+            ascending order for a given row.
+          | For COO this guarantees that that the row indices are sorted in
+            ascending order.
```

Comment:
do we need column indices to also be sorted within the row sets for COO format?

Reply:
This is a question for @gajanan-choudhary.

Reply:
Yes, column indices should also be sorted. In Excel speak, that's "Sort by `rowptr` then by `colind`". If we were okay with COO column indices not being sorted, we wouldn't have a `sorted` property for `CSR` matrices. If we have to deal with unsorted COO column indices some day, we can pick a new enum name and add it in.

Reply:
I clarified this in https://github.com/uxlfoundation/oneAPI-spec/pull/522/commits/669a4e7a58796672124662b3010e62ea1e3e5c22